File: 9.c
// Printing Primes till 50000 (Recursion)
#include <stdio.h>
int PrimeCheck(int start, int end);
void Increment(int i);
int main()
{
    Increment(0);
    return 0;
}
int PrimeCheck(int start, int end)
{
    if (start < end)
    {
        if (end % start == 0)
        {
            return -1;
        }
        return PrimeCheck(start + 1, end);
    }
    return 0;
}

void Increment(int i)
{
    if (i <= 50000)
    {
        if (PrimeCheck(2, i) == 0)
        {
            printf("%d\n", i);
        }
        return Increment(i + 1);
    }
}

File: 9.js
// Printing Primes till 50000 (Recursion)
function PrimeCheck(start, end) {
    if (start < end) {
        if (end % start == 0) {
            return -1;
        }
        return PrimeCheck(start + 1, end);
    }
    return 0;
}

function Increment(i) {
    if (i <= 50000) {
        if (PrimeCheck(2, i) == 0) {
            console.log(i);
        }
        return Increment(i + 1);
    }
}

Increment(0);